README.md

```diff
@@ -17,3 +17,329 @@ Implement a prototype job worker service that provides an API to run arbitrary L
+}
+
+type QueryPidResponse {
+    LogOutput output 
+    LogOutput errorOutput 
+    bool isRunning 
+}
+
+type QueryRunningProcessesRequest {
+    //will be empty since the server just needs to verify its a QueryRunningProcessesRequest
+}
+    
+type QueryRunningProcessesResponse {
+     ProcessInfo[] processTable = 1;
+}
+
+type QueryListOfLogsRequest {
+    //will be empty since the server just needs to verify its a QueryListOfLogsRequest
+}
+
+type QueryListOfLogsResponse {
+    repeated ProcessInfo processTable = 1;
+}
+
+type QueryRunningProcessesResponse {
+     ProcessInfo[] processTable = 1;
+}
+type QueryRequest {
+    variableField queryCommand {
+        QueryPidRequest queryPid 
+        QueryRunningProcessesRequest queryRunningProcesses 
+        QueryListOfLogsResponse responseFromListOfLogs
+    }
+}
+
+type QueryResponse {
+variableField responseFromQuery {
+        QueryPidResponse responseFromQueryPid
+        QueryRunningProcessesResponse responseFromRunningProcesses
+        QueryListOfLogsResponse responseFromListOfLogs 
+    }
+}
+
+```
+
+
+### Implementation Overview 
+
+* Using GRPC so the above methods and types with be generated via protocol buffers (will be using libprotoc 3.11.4)
+
+####  DataStore
+
+* we can use sync.Map in Go to implement a set of structs to store process info
+
+
+#### Server 
+
+* data store for running pid, starting timestamp, process names
+    
+    * can be implemented with a concurrent map
+
+* function to execute commands 
+
+    * os/exec and syscall packages can take care of this
+
+* logfile system management
+
+    * can use log, writer and reader packages
+
+    * must manage concurrency issues with read and write
+    
+    * will be similar to how linux implements it using the /proc path, but simpler:
+        ```
+        $ ls -al  /proc | head -n 10
+        total 4
+        dr-xr-xr-x 312 root             root                           0 Aug 24 11:41 .
+        drwxr-xr-x  23 root             root                        4096 Aug 17 16:01 ..
+        dr-xr-xr-x   9 root             root                           0 Aug 24 11:41 1
+        dr-xr-xr-x   9 root             root                           0 Aug 26 16:40 10
+        dr-xr-xr-x   9 avwong13         avwong13                       0 Aug 26 16:40 1003
+        dr-xr-xr-x   9 root             root                           0 Aug 26 17:08 1004
+        dr-xr-xr-x   9 avwong13         avwong13                       0 Aug 25 23:08 10202
+        dr-xr-xr-x   9 avwong13         avwong13                       0 Aug 25 23:09 10256
+        dr-xr-xr-x   9 avwong13         avwong13                       0 Aug 24 13:25 10485
+        ```
+
+* The server will only store running pids in memory and have the pass logs available in their respective \<Pid\>_\<startingTimeStamp\> folder
+
+* concurrent go routines to handle cmd executions starting and finishing 
+
+* killing of pid can be implemented can be done through sigkill via syscall
```

Comment:
use https://golang.org/pkg/os/#Process.Signal instead (`os.Process` is provided by `os/exec.Command`)